product_code/authentication_service/impl/src/main/scala/de/upb/cs/uc4/authentication/impl/AuthenticationServiceImpl.scala

```diff
@@ -51,19 +51,35 @@ class AuthenticationServiceImpl(readSide: ReadSide, processor: AuthenticationEve
   override def changePassword(username: String): ServiceCall[AuthenticationUser, Done] =
     identifiedAuthenticated[AuthenticationUser, Done](AuthenticationRole.All: _*) {
       (authUsername, role) =>
-        ServerServiceCall { (_: RequestHeader, user: AuthenticationUser) =>
-          if (username != user.username.trim) {
+        ServerServiceCall { (_: RequestHeader, authUserRaw: AuthenticationUser) =>
+          val authUser = authUserRaw.clean
+
+          if (username != authUser.username) {
             throw UC4Exception.PathParameterMismatch
           }
-          if (authUsername != user.username.trim) {
+          if (authUsername != authUser.username) {
             throw UC4Exception.OwnerMismatch
           }
-          if (role != user.role) {
+
+          val validationErrors = try {
+            Await.result(authUser.validate, 5.seconds)
+          }
+          catch {
+            case _: TimeoutException => throw UC4Exception.ValidationTimeout
+            case e: Exception        => throw e
```

Comment:
This needs to be changed into a Internal Server Error
```suggestion
            case e: Exception        => throw UC4Exception.InternalServerError("Validation Error", e.getMessage)
```